src/main/java/apoc/export/json/JsonFormat.java

```diff
@@ -0,0 +1,139 @@
+import java.io.Writer;
+import java.util.Iterator;
+import java.util.List;
+
+import static apoc.util.JsonUtil.getPropertiesArray;
+
+public class JsonFormat implements Format {
+    private final GraphDatabaseService db;
+
+    public JsonFormat(GraphDatabaseService db) {
+        this.db = db;
+    }
+
+    @Override
+    public ProgressInfo load(Reader reader, Reporter reporter, ExportConfig config) throws Exception {
+        return null;
+    }
+
+    @Override
+    public ProgressInfo dump(SubGraph graph, Writer writer, Reporter reporter, ExportConfig config) throws Exception {
+        try (Transaction tx = db.beginTx()) {
+            writeAll(graph, reporter, config, writer);
+            tx.success();
+            writer.close();
+            return reporter.getTotal();
+        }
+    }
+
+    public ProgressInfo dump(Result result, Writer writer, Reporter reporter, ExportConfig config) throws Exception {
+        try (Transaction tx = db.beginTx()) {
+            ObjectMapper mapperNode = new ObjectMapper();
+            ObjectMapper mapperRel = new ObjectMapper();
+            ObjectMapper mapperProp = new ObjectMapper();
+
+            List<String> column = result.columns();
+            String[] header = column.toArray(new String[result.columns().size()]);
+
+            SimpleModule module = new SimpleModule("CustomSerializeNode", new Version(1, 0, 0, null, null, null));
+            module.addSerializer(Node.class, new CustomSerializeNode());
+
+            SimpleModule moduleRel = new SimpleModule("CustomSerializeRelationship", new Version(1, 0, 0, null, null, null));
+            moduleRel.addSerializer(Relationship.class, new CustomSerializeRelationship());
+
+            mapperNode.registerModule(module);
+            mapperRel.registerModule(moduleRel);
+
+            writer.write("[ ");
+            result.accept((row) -> {
+                ObjectNode objectNode = mapperProp.createObjectNode();
+                String node = "", rel = "", property = "";
+                for (int col = 0; col < header.length; col++) {
+                    Object value = row.get(header[col]);
+                    if (value instanceof Node) {
+                        node = mapperNode.writerWithDefaultPrettyPrinter().writeValueAsString(value);
+                    }
+                    if (value instanceof Relationship) {
+                        rel = mapperRel.writerWithDefaultPrettyPrinter().writeValueAsString(value);
+                    }
+                    if (!(value instanceof PropertyContainer)) {
+                        if(value != null) {
+                            property = getPropertiesArray(mapperProp, header[col], value, objectNode);
+                        }
+                    }
+                    reporter.update(value instanceof Node ? 1 : 0, value instanceof Relationship ? 1 : 0, value instanceof PropertyContainer ? 0 : 1);
+                }
+                writeJsonToFile(writer, property, result.hasNext());

```

Comment:
Should be inside `if (value instanceof Node) ` otherwise with multiple nodes returned the `node` variablewill be overwritten